checks.d/mapreduce.py

```diff
@@ -0,0 +1,481 @@
+
+# Service Check Names
+SERVICE_CHECK = 'mapreduce.cluster.can_connect'
+YARN_SERVICE_CHECK = 'mapreduce.resource_manager.can_connect'
+MAPREDUCE_SERVICE_CHECK = 'mapreduce.application_master.can_connect'
+
+# URL Paths
+INFO_PATH = 'ws/v1/cluster/info'
+YARN_APPS_PATH = 'ws/v1/cluster/apps'
+MAPREDUCE_JOBS_PATH = 'ws/v1/mapreduce/jobs'
+
+
+TASK_COUNTER_GROUP_NAME = 'org.apache.hadoop.mapreduce.TaskCounter'
+
+# Application type and states to collect
+YARN_APPLICATION_TYPES = 'MAPREDUCE'
+YARN_APPLICATION_STATES = 'RUNNING'
+
+# Metric types
+GAUGE = 'gauge'
+
+# Metrics to collect
+MAPREDUCE_JOB_METRICS = {
+    'elapsedTime': ('mapreduce.job.elapsed_time', GAUGE),
+    'mapsTotal': ('mapreduce.job.maps_total', GAUGE),
+    'mapsCompleted': ('mapreduce.job.maps_completed', GAUGE),
+    'reducesTotal': ('mapreduce.job.reduces_total', GAUGE),
+    'reducesCompleted': ('mapreduce.job.reduces_completed', GAUGE),
+    'mapsPending': ('mapreduce.job.maps_pending', GAUGE),
+    'mapsRunning': ('mapreduce.job.maps_running', GAUGE),
+    'reducesPending': ('mapreduce.job.reduces_pending', GAUGE),
+    'reducesRunning': ('mapreduce.job.reduces_running', GAUGE),
+    'newReduceAttempts': ('mapreduce.job.new_reduce_attempts', GAUGE),
+    'runningReduceAttempts': ('mapreduce.job.running_reduce_attempts', GAUGE),
+    'failedReduceAttempts': ('mapreduce.job.failed_reduce_attempts', GAUGE),
+    'killedReduceAttempts': ('mapreduce.job.killed_reduce_attempts', GAUGE),
+    'successfulReduceAttempts': ('mapreduce.job.successful_reduce_attempts', GAUGE),
+    'newMapAttempts': ('mapreduce.job.new_map_attempts', GAUGE),
+    'runningMapAttempts': ('mapreduce.job.running_map_attempts', GAUGE),
+    'failedMapAttempts': ('mapreduce.job.failed_map_attempts', GAUGE),
+    'killedMapAttempts': ('mapreduce.job.killed_map_attempts', GAUGE),
+    'successfulMapAttempts': ('mapreduce.job.successful_map_attempts', GAUGE),
+}
+
+MAPREDUCE_JOB_COUNTER_METRICS = {
+    'reduceCounterValue': ('mapreduce.job.counter.reduce_counter_value', GAUGE),
+    'mapCounterValue': ('mapreduce.job.counter.map_counter_value', GAUGE),
+    'totalCounterValue': ('mapreduce.job.counter.total_counter_value', GAUGE),
+}
+
+MAPREDUCE_TASK_METRICS = {
+    'progress': ('mapreduce.job.task.progress', GAUGE),
+    'elapsedTime': ('mapreduce.job.task.elapsed_time', GAUGE)
+}
+
+MAPREDUCE_TASK_STATE_METRICS = {
+    'NEW': ('maprduce.job.task.state.new', GAUGE),
+    'SCHEDULED': ('maprduce.job.task.state.scheduled', GAUGE),
+    'RUNNING': ('maprduce.job.task.state.running', GAUGE),
+    'SUCCEEDED': ('maprduce.job.task.state.succeeded', GAUGE),
+    'FAILED': ('maprduce.job.task.state.failed', GAUGE),
+    'KILL_WAIT': ('maprduce.job.task.state.killWait', GAUGE),
+    'KILLED': ('maprduce.job.task.state.killed', GAUGE),
+}
+
+MAPREDUCE_TASK_TYPE_METRICS = {
+    'MAP': ('maprduce.job.task.type.map', GAUGE),
+    'REDUCE': ('maprduce.job.task.type.reduce', GAUGE),
+}
+
```

Comment:
Some typo here, `MAPREDUCE_TASK_METRICS ` & `MAPREDUCE_TASK_STATE_METRICS ` metrics are namespaced with `maprduce ` instead of `mapreduce` :pencil2: .